sdmetrics/column_pairs/statistical/correlation_similarity.py

```diff
@@ -0,0 +1,97 @@
+"""Correlation Similarity Metric."""
+
+import pandas as pd
+from scipy.stats import pearsonr, spearmanr
+
+from sdmetrics.column_pairs.base import ColumnPairsMetric
+from sdmetrics.goal import Goal
+from sdmetrics.utils import is_datetime
+
+
+class CorrelationSimilarity(ColumnPairsMetric):
+    """Correlation similarity metric.
+
+    Attributes:
+        name (str):
+            Name to use when reports about this metric are printed.
+        goal (sdmetrics.goal.Goal):
+            The goal of this metric.
+        min_value (Union[float, tuple[float]]):
+            Minimum value or values that this metric can take.
+        max_value (Union[float, tuple[float]]):
+            Maximum value or values that this metric can take.
+    """
+
+    name = 'CorrelationSimilarity'
+    goal = Goal.MAXIMIZE
+    min_value = 0.0
+    max_value = 1.0
+
+    @classmethod
+    def compute_breakdown(cls, real_data, synthetic_data, coefficient='Pearson'):
+        """Compare the breakdown of correlation similarity of two continuous columns.
+
+        Args:
+            real_data (Union[numpy.ndarray, pandas.Series]):
+                The values from the real dataset.
+            synthetic_data (Union[numpy.ndarray, pandas.Series]):
+                The values from the synthetic dataset.
+
+        Returns:
+            dict:
+                A dict containing the score, and the real and synthetic metric values.
+        """
+        real_data[pd.isna(real_data)] = 0.0
+        synthetic_data[pd.isna(synthetic_data)] = 0.0
+        column1, column2 = real_data.columns[:2]
+
+        if is_datetime(real_data):
```

Comment:
Hm so I have a couple thoughts - 
1. This handling is used in many metrics across the `column_pairs` and `single_column` metrics. I do think it would be nice to add more comprehensive validation. I'm not sure if it's worth adding it in this PR, since that will make it less unified.
2. Most of the usage should be through single table metrics, which filters all the metrics by data type, and only applies the relevant metrics. In this use case, we should always be applying the correct metric for a given data type. That's why I don't think it's a huge issue right now. Of course, people can still choose to manually invoke the column pair metric.

I'm in favor of opening another issue around adding data type verification on the base classes of `ColumnPairMetric` and `SingleColumnMetric`, and addressing it for all metrics. Let me know what you think.

Reply:
Makes sense to me

Reply:
Cool, added an issue for tracking here: https://github.com/sdv-dev/SDMetrics/issues/168